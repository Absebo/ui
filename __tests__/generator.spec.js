import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Generator } from '../src/index.js';

describe('generator', () => {
  test('initial render remains consistent', () => {
    const wrapper = mount(
      <Generator isOpen={true}>
        <a />
      </Generator>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
