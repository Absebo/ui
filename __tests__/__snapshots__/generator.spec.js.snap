// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator initial render remains consistent 1`] = `
<Generator
  isOpen={true}
  onGenerate={[Function]}
>
  <Styled(Popover)
    body={
      <styled.div
        className={undefined}
      >
        <styled.pre
          onClick={[Function]}
          role="content"
        >
          <Styled(ColoredDigits)
            value=""
          />
        </styled.pre>
        <styled.div>
          <styled.div>
            <input
              checked={true}
              onChange={[Function]}
              type="radio"
            />
             
            Characters
          </styled.div>
          <styled.div>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
            />
             
            Words
          </styled.div>
        </styled.div>
        <styled.fieldset>
          <styled.legend>
            Options
          </styled.legend>
          <styled.div>
            <input
              max="50"
              min="10"
              onChange={[Function]}
              type="range"
              value={20}
            />
            <span>
              20
            </span>
          </styled.div>
          <styled.div>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox"
            />
             
            Uppercase Letters
          </styled.div>
          <styled.div>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox"
            />
             
            Lowercase Letters
          </styled.div>
          <styled.div>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox"
            />
             
            Digits
          </styled.div>
          <styled.div>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
             
            Space
          </styled.div>
          <styled.div>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox"
            />
             
            Underscore & Dash
          </styled.div>
          <styled.div>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
             
            Symbols
          </styled.div>
        </styled.fieldset>
        <styled.div>
          <Styled(BaseButton)
            onClick={[Function]}
            primary={true}
          >
            Generate
          </Styled(BaseButton)>
          <Styled(BaseButton)
            dark={true}
            onClick={[Function]}
          >
            Use This
          </Styled(BaseButton)>
        </styled.div>
      </styled.div>
    }
    isOpen={true}
    onGenerate={[Function]}
  >
    <Popover
      appendTarget={
        <body>
          <div
            class="Popover Popover-above sc-ifAKCX bhoBrh"
            style="position: absolute; top: -4px; left: -25px; display: flex; opacity: 1;"
          >
            <div
              class="Popover-body"
            >
              <div
                class="sc-bZQynM dHqpWy"
              >
                <pre
                  class="sc-gzVnrw hYukHf"
                  role="content"
                >
                  <span
                    class="sc-EHOje jFNwUJ"
                  >
                    <span
                      class="str"
                    />
                  </span>
                </pre>
                <div
                  class="sc-htoDjs bcFUFu"
                >
                  <div
                    class="sc-dnqmqq jPiRWc"
                  >
                    <input
                      type="radio"
                      value="on"
                    />
                     
                    Characters
                  </div>
                  <div
                    class="sc-dnqmqq jPiRWc"
                  >
                    <input
                      type="radio"
                      value="on"
                    />
                     
                    Words
                  </div>
                </div>
                <fieldset
                  class="sc-gZMcBi eQdbrd"
                >
                  <legend
                    class="sc-gqjmRU hyAZFF"
                  >
                    Options
                  </legend>
                  <div
                    class="sc-iwsKbI cPIsdT"
                  >
                    <input
                      max="50"
                      min="10"
                      type="range"
                      value="20"
                    />
                    <span>
                      20
                    </span>
                  </div>
                  <div
                    class="sc-dnqmqq jPiRWc"
                  >
                    <input
                      type="checkbox"
                      value="on"
                    />
                     
                    Uppercase Letters
                  </div>
                  <div
                    class="sc-dnqmqq jPiRWc"
                  >
                    <input
                      type="checkbox"
                      value="on"
                    />
                     
                    Lowercase Letters
                  </div>
                  <div
                    class="sc-dnqmqq jPiRWc"
                  >
                    <input
                      type="checkbox"
                      value="on"
                    />
                     
                    Digits
                  </div>
                  <div
                    class="sc-dnqmqq jPiRWc"
                  >
                    <input
                      type="checkbox"
                      value="on"
                    />
                     
                    Space
                  </div>
                  <div
                    class="sc-dnqmqq jPiRWc"
                  >
                    <input
                      type="checkbox"
                      value="on"
                    />
                     
                    Underscore & Dash
                  </div>
                  <div
                    class="sc-dnqmqq jPiRWc"
                  >
                    <input
                      type="checkbox"
                      value="on"
                    />
                     
                    Symbols
                  </div>
                </fieldset>
                <div
                  class="sc-VigVT dpurnc"
                >
                  <button
                    class="sc-bdVaJa cuAuCI"
                    type="button"
                  >
                    Generate
                  </button>
                  <button
                    class="sc-bdVaJa ihgpPH"
                    type="button"
                  >
                    Use This
                  </button>
                </div>
              </div>
            </div>
            <svg
              class="Popover-tip"
              height="7"
              width="14"
            >
              <polygon
                class="Popover-tipShape"
                points="0,0 7,7, 14,0"
              />
            </svg>
            <object
              class="resize-sensor"
              data="about:blank"
              style="display: block; position: absolute; top: 0; left: 0; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
              type="text/html"
            />
          </div>
        </body>
      }
      body={
        <styled.div
          className={undefined}
        >
          <styled.pre
            onClick={[Function]}
            role="content"
          >
            <Styled(ColoredDigits)
              value=""
            />
          </styled.pre>
          <styled.div>
            <styled.div>
              <input
                checked={true}
                onChange={[Function]}
                type="radio"
              />
               
              Characters
            </styled.div>
            <styled.div>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
              />
               
              Words
            </styled.div>
          </styled.div>
          <styled.fieldset>
            <styled.legend>
              Options
            </styled.legend>
            <styled.div>
              <input
                max="50"
                min="10"
                onChange={[Function]}
                type="range"
                value={20}
              />
              <span>
                20
              </span>
            </styled.div>
            <styled.div>
              <input
                checked={true}
                onChange={[Function]}
                type="checkbox"
              />
               
              Uppercase Letters
            </styled.div>
            <styled.div>
              <input
                checked={true}
                onChange={[Function]}
                type="checkbox"
              />
               
              Lowercase Letters
            </styled.div>
            <styled.div>
              <input
                checked={true}
                onChange={[Function]}
                type="checkbox"
              />
               
              Digits
            </styled.div>
            <styled.div>
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />
               
              Space
            </styled.div>
            <styled.div>
              <input
                checked={true}
                onChange={[Function]}
                type="checkbox"
              />
               
              Underscore & Dash
            </styled.div>
            <styled.div>
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />
               
              Symbols
            </styled.div>
          </styled.fieldset>
          <styled.div>
            <Styled(BaseButton)
              onClick={[Function]}
              primary={true}
            >
              Generate
            </Styled(BaseButton)>
            <Styled(BaseButton)
              dark={true}
              onClick={[Function]}
            >
              Use This
            </Styled(BaseButton)>
          </styled.div>
        </styled.div>
      }
      className="sc-ifAKCX bhoBrh"
      enterExitTransitionDurationMs={500}
      isOpen={true}
      offset={4}
      onGenerate={[Function]}
      onOuterAction={[Function]}
      place={null}
      preferPlace={null}
      refreshIntervalMs={200}
      tipSize={7}
    >
      <a />
      <div
        className="Popover Popover-above sc-ifAKCX bhoBrh"
        style={
          Object {
            "display": "flex",
            "left": 0,
            "position": "absolute",
            "top": 0,
          }
        }
      >
        <div
          className="Popover-body"
        >
          <styled.div>
            <div
              className="sc-bZQynM dHqpWy"
            >
              <styled.pre
                onClick={[Function]}
                role="content"
              >
                <pre
                  className="sc-gzVnrw hYukHf"
                  onClick={[Function]}
                  role="content"
                >
                  <Styled(ColoredDigits)
                    value=""
                  >
                    <ColoredDigits
                      className="sc-EHOje jFNwUJ"
                      value=""
                    >
                      <span
                        className="sc-EHOje jFNwUJ"
                      >
                        <span
                          className="str"
                          key="0"
                        />
                      </span>
                    </ColoredDigits>
                  </Styled(ColoredDigits)>
                </pre>
              </styled.pre>
              <styled.div>
                <div
                  className="sc-htoDjs bcFUFu"
                >
                  <styled.div>
                    <div
                      className="sc-dnqmqq jPiRWc"
                    >
                      <input
                        checked={true}
                        onChange={[Function]}
                        type="radio"
                      />
                       
                      Characters
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="sc-dnqmqq jPiRWc"
                    >
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="radio"
                      />
                       
                      Words
                    </div>
                  </styled.div>
                </div>
              </styled.div>
              <styled.fieldset>
                <fieldset
                  className="sc-gZMcBi eQdbrd"
                >
                  <styled.legend>
                    <legend
                      className="sc-gqjmRU hyAZFF"
                    >
                      Options
                    </legend>
                  </styled.legend>
                  <styled.div>
                    <div
                      className="sc-iwsKbI cPIsdT"
                    >
                      <input
                        max="50"
                        min="10"
                        onChange={[Function]}
                        type="range"
                        value={20}
                      />
                      <span>
                        20
                      </span>
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="sc-dnqmqq jPiRWc"
                    >
                      <input
                        checked={true}
                        onChange={[Function]}
                        type="checkbox"
                      />
                       
                      Uppercase Letters
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="sc-dnqmqq jPiRWc"
                    >
                      <input
                        checked={true}
                        onChange={[Function]}
                        type="checkbox"
                      />
                       
                      Lowercase Letters
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="sc-dnqmqq jPiRWc"
                    >
                      <input
                        checked={true}
                        onChange={[Function]}
                        type="checkbox"
                      />
                       
                      Digits
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="sc-dnqmqq jPiRWc"
                    >
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                      />
                       
                      Space
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="sc-dnqmqq jPiRWc"
                    >
                      <input
                        checked={true}
                        onChange={[Function]}
                        type="checkbox"
                      />
                       
                      Underscore & Dash
                    </div>
                  </styled.div>
                  <styled.div>
                    <div
                      className="sc-dnqmqq jPiRWc"
                    >
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                      />
                       
                      Symbols
                    </div>
                  </styled.div>
                </fieldset>
              </styled.fieldset>
              <styled.div>
                <div
                  className="sc-VigVT dpurnc"
                >
                  <Styled(BaseButton)
                    onClick={[Function]}
                    primary={true}
                  >
                    <BaseButton
                      className="sc-bdVaJa cuAuCI"
                      onClick={[Function]}
                      primary={true}
                    >
                      <button
                        className="sc-bdVaJa cuAuCI"
                        onClick={[Function]}
                        type="button"
                      >
                        Generate
                      </button>
                    </BaseButton>
                  </Styled(BaseButton)>
                  <Styled(BaseButton)
                    dark={true}
                    onClick={[Function]}
                  >
                    <BaseButton
                      className="sc-bdVaJa ihgpPH"
                      dark={true}
                      onClick={[Function]}
                    >
                      <button
                        className="sc-bdVaJa ihgpPH"
                        onClick={[Function]}
                        type="button"
                      >
                        Use This
                      </button>
                    </BaseButton>
                  </Styled(BaseButton)>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
        <Tip
          direction="down"
          size={7}
        >
          <svg
            className="Popover-tip"
            height={7}
            width={14}
          >
            <polygon
              className="Popover-tipShape"
              points="0,0 7,7, 14,0"
            />
          </svg>
        </Tip>
      </div>
    </Popover>
  </Styled(Popover)>
</Generator>
`;
